sim/task.h

```diff
@@ -65,11 +70,11 @@ typedef void (*TaskFunction_t)(void *instance);
  * \ingroup Tasks
  */
 //typedef void * TaskHandle_t;
-struct TaskHandle_t {
-  void *thread_handle = nullptr;
+typedef struct TaskHandle_t {
+  void *thread_handle;
```

Comment:
can we still default initialize in `C` code? I actually don't know

```suggestion
  void *thread_handle = 0;
```

Reply:
Mhm I don't think we can do that in `C`, but this file should be compiled by the C++ compiler so it should work, I guess (and it builds).